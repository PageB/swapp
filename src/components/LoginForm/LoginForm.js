import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Button, Input } from '../../components';
import styles from './LoginForm.module.scss';

class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.state = { email: '', password: '', errorMessage: this.props.error };
  }

  /**
   * Update corresponding input on change.
   *
   * @method handleInputChange
   * @param {Object} event
   */
  handleInputChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  /**
   * Call login action on form submit.
   *
   * @method handleInputChange
   * @param {Object} event
   */
  handleSubmit = e => {
    e.preventDefault();
    this.props.login({ variables: { email: this.state.email, password: this.state.password } });
  };

  /**
   * Render lifecycle hook.
   *
   * @method render
   */
  render() {
    const { email, password, errorMessage } = this.state;

    const LoginErrorMessage = errorMessage ? (
      <p className={styles.ErrorMessage}>{errorMessage}</p>
    ) : (
      <></>
    );

    return (
      <form
        className={[styles.Form, styles[this.props.theme]].join(' ')}
        onSubmit={this.handleSubmit}
      >
        {LoginErrorMessage}
        <Input
          type="text"
          placeholder="Email"
          name="email"
          value={email}
          autocomplete="on"
          onChange={this.handleInputChange}
        />
        <Input
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          autocomplete="on"
          onChange={this.handleInputChange}
        />
        <Button type="submit" theme={this.props.theme}>
          Login
        </Button>
      </form>
    );
  }
}

LoginForm.propTypes = {
  theme: PropTypes.oneOf(['DarkTheme', 'LightTheme']),
};

LoginForm.defaultProps = {
  theme: 'DarkTheme',
};

export default LoginForm;
