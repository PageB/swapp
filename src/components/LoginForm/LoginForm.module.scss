@import '../../colors.scss';

.DarkTheme {
  background-color: #333;
}

.LightTheme {
  background-color: white;
}

.Form {
  width: 420px;
  margin: 0 auto;
  padding: 56px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  border-radius: 8px;

  input {
    margin-bottom: 8px;
  }
}

.ErrorMessage {
  color: #e27164;
  font-weight: bold;
  align-self: flex-start;
  padding: 0;
  margin: 0;
  margin-bottom: 8px;
}
