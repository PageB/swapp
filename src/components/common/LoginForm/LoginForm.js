import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import Input from '../Input/Input';
import { AUTH } from '../../../constants';

import styles from './LoginForm.module.scss';

class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.state = { email: '', password: '', errorMessage: this.props.errorMessage };
  }

  handleInputChange = ({ currentTarget }) => {
    this.setState({ [currentTarget.name]: currentTarget.value });
  };

  handleSubmit = e => {
    e.preventDefault();

    const url = 'http://softuni-swapp-212366186.eu-west-1.elb.amazonaws.com/graphql';
    const options = {
      method: 'POST',
      headers: {
        Authentication: '',
      },
      body: JSON.stringify({
        mutation: `{ signIn(email: ${this.state.email} password: ${this.state.password}) }`,
      }),
    };

    fetch(url, options)
      .then(response => {
        debugger;
        if (!response.ok) {
          if (response.status === 404) {
            alert('Email not found, please retry');
          }
          if (response.status === 401) {
            alert('Email and password do not match, please retry');
          }
        }
        return response;
      })
      .then(response => response.json())
      .then(data => {
        if (data.success) {
          document.cookie = 'token=' + data.token;
          // navigate('/private-area')
        }
      });
  };

  handleeUserData = token => {
    if (token) {
      localStorage.removeItem(AUTH.AUTH_TOKEN);
      // this.props.history.push(`/`);
    } else {
      localStorage.setItem(AUTH.AUTH_TOKEN, token);
    }
  };

  render() {
    const { email, password, errorMessage } = this.state;

    const LoginErrorMessage = errorMessage ? (
      <p className={styles.ErrorMessage}>{errorMessage}</p>
    ) : (
      <p></p>
    );

    return (
      <form
        className={[styles.Form, styles[this.props.theme]].join(' ')}
        onSubmit={this.handleSubmit}
      >
        {LoginErrorMessage}
        <Input
          type="text"
          placeholder="Email"
          name="email"
          value={email}
          onChange={this.handleInputChange}
        />
        <Input
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          onChange={this.handleInputChange}
        />
        <Button type="submit" theme={this.props.theme}>
          Login
        </Button>
      </form>
    );
  }
}

LoginForm.propTypes = {
  theme: PropTypes.oneOf(['DarkTheme', 'LightTheme']),
};

LoginForm.defaultProps = {
  theme: 'DarkTheme',
};

export default LoginForm;
