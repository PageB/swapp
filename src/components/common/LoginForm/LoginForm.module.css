.Form {
  width: 420px;
  margin: 0 auto;
  padding: 56px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  background-color: #3c4858;
  border: 1px solid #333333;
  border-radius: 8px;
}

.ErrorMessage {
  color: #e27164;
  font-weight: bold;
  align-self: flex-start;
  padding: 0px;
  margin: 0px;
  margin-bottom: 8px;
}

.Form input {
  margin-bottom: 8px;
}
