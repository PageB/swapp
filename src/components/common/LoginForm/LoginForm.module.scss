@import '../../../colors.scss';

.Form {
  width: 420px;
  margin: 0 auto;
  padding: 56px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  border-radius: 8px;
}

.ErrorMessage {
  color: #e27164;
  font-weight: bold;
  align-self: flex-start;
  padding: 0px;
  margin: 0px;
  margin-bottom: 8px;
}

.Form input {
  margin-bottom: 8px;
}

.DarkTheme {
  background-color: #333;
}

.LightTheme {
  background-color: white;
}
