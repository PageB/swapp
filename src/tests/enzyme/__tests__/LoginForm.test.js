import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, mount } from 'enzyme';

import LoginForm from '../../../components/LoginForm/LoginForm';
import { setTimeout } from 'timers';

configure({ adapter: new Adapter() });

describe('LoginForm Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<LoginForm />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call login when form is submited', () => {
    const preventDefault = jest.fn();
    const login = jest.fn();
    const wrapper = mount(<LoginForm login={login} />);

    wrapper.find('form').simulate('submit', { preventDefault });

    expect(preventDefault).toHaveBeenCalledTimes(1);
    expect(login).toHaveBeenCalledTimes(1);
  });

  it('should change input value on change event', () => {
    const wrapper = mount(<LoginForm />);

    wrapper
      .find('input')
      .first()
      .simulate('focus');
    wrapper
      .find('input')
      .first()
      .simulate('change', { target: { value: 'First line', name: 'email' } });

    expect(
      wrapper
        .find('input')
        .first()
        .instance().value,
    ).toBe('First line');
  });

  it('should show error message', () => {
    const wrapper = mount(<LoginForm error={'Incorrect email or password.'} />);

    expect(wrapper.find('.ErrorMessage').text()).toBe('Incorrect email or password.');
  });
});
