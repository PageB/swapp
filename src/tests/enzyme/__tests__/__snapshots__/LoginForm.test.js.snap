// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should match snapshot 1`] = `
<form
  className="Form DarkTheme"
  onSubmit={[Function]}
>
  <p />
  <Input
    autocomplete="on"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <Input
    autocomplete="on"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <Button
    theme="DarkTheme"
    type="submit"
  >
    Login
  </Button>
</form>
`;
